cli/src/description_util.rs

```diff
@@ -318,15 +319,46 @@ pub fn try_combine_messages(sources: &[Commit], destination: &Commit) -> Option<
 ///
 /// This includes empty descriptins too, so the user doesn't have to wonder why
 /// they only see 2 descriptions when they combined 3 commits.
-pub fn combine_messages_for_editing(sources: &[Commit], destination: &Commit) -> String {
+pub fn combine_messages_for_editing(
+    ui: &Ui,
+    tx: &WorkspaceCommandTransaction,
+    sources: &[Commit],
+    destination: &Commit,
+    commit_builder: &DetachedCommitBuilder,
+) -> Result<String, CommandError> {
     let mut combined = String::new();
     combined.push_str("JJ: Description from the destination commit:\n");
     combined.push_str(destination.description());
     for commit in sources {
         combined.push_str("\nJJ: Description from source commit:\n");
         combined.push_str(commit.description());
     }
-    combined
+    if let Some(template) = parse_trailers_template(ui, tx)? {
+        let old_trailers: HashSet<_> = sources
+            .iter()
+            .chain(std::iter::once(destination))
+            .flat_map(|commit| parse_description_trailers(commit.description()))
+            .collect();
+        let commit = commit_builder.write_hidden()?;
+        let mut output = Vec::new();
+        template
+            .format(&commit, &mut PlainTextFormatter::new(&mut output))
+            .expect("write() to vec backed formatter should never fail");
+        let trailer_lines = output
+            .into_string()
+            .map_err(|_| user_error("Trailers should be valid utf-8"))?;
+        let new_trailers: HashSet<_> = parse_trailers(&trailer_lines)?.into_iter().collect();
```

Comment:
Avoid using `HashSet` for `new_trailers` to stabilize order. `.filter()` can be used instead.